### ISSUE TITLE
Accessory being deleted/recreated when Homebridge is restarted

### ISSUE THREAD

**Issue by Djelibeybi** [ebaauw/homebridge-soma#8]
~I haven't traced the code to be 100% sure, but it seems to me that `homebridge-soma` [deletes the accessory in the event of an error](https://github.com/ebaauw/homebridge-soma/blob/6f23a62860b3535c7ab6ab170316074007eec82e/lib/SomaPlatform.js#L120-L136).~

~I'm seeing this happen fairly regularly, I suspect because of the [RangeError](https://github.com/ebaauw/homebridge-soma/issues/6), which means the blinds disappear from HomeKit then reappear (after I restart Homebridge) as new accessories, breaking any existing automation and reverting back to the name given in the Smart Shades app which is not ideal, because the app has limit on the length of the name and it's tiny~ 

~Can you perhaps make it slightly more tolerant of transient errors by flagging the accessory as offline, rather than removing it? I seem to recall that there was a way to do this with Homebridge/HAP-NodeJS, but I also have this feeling the method is deprecated or no longer supported.~

Whenever I restart Homebridge, it removes the accessory as "stale" and then creates a new one.

**Comment by Djelibeybi** [comment]
I just noticed this in the logs after restarting Homebridge:

```
[01/11/2021, 09:09:46] [SOMA] Avi's Office: remove stale SomaAccessory v1.0.3 accessory CD788D2CC7D8
```

Not sure why it thought the accessory was stale though. Perhaps a whitelisting bug?

**Comment by ebaauw** [comment]
Are you whitelisting by name or by mac address?

**Comment by Djelibeybi** [comment]
By MAC address.

**Comment by ebaauw** [comment]
My bad.  Comparing the ID (without `:`s) against the MAC address (with `:`s) doesn't work, obviously.

Fixed in v1.0.4.  Note that the first time you start v1.0.4, the devices will still be removes, as v1.0.3 doesn't persist the MAC address, but after the first restart in v1.0.4 it should work.

**Comment by Djelibeybi** [comment]
No problem. I have moved all my automations to Node-RED so this was more of annoyance for me. Glad it was an easy fix though.

**Comment by Djelibeybi** [comment]
Confirmed fixed with 1.0.4.
